src/bytes.js

```diff
@@ -0,0 +1,1443 @@
+require("fastestsmallesttextencoderdecoder");
+
+// Mapping from supported valid encodings to normalized encoding name
+const supportedEncodings = {
+    "utf": "utf-8",
+    "utf8": "utf-8",
+    "utf-8": "utf-8",
+    "ascii": "ascii"
+};
+
+function normalizeEncoding (encoding) {
+    const normalized = encoding.replace(/\s+/g, "").toLowerCase();
+    const supported = supportedEncodings[normalized];
+    if (supported === undefined) {
+        return encoding;
+    } else {
+        return supported;
+    }
+}
+
+/**
+ * @constructor
+ * @param {*} source
+ * @param {Sk.builtin.str=} encoding
+ * @param {Sk.builtin.str=} errors
+ * @return {Sk.builtin.bytes}
+ */
+
+Sk.builtin.bytes = function (source, encoding, errors) {
+    let view;
+    let arr;
+
+    if (!(this instanceof Sk.builtin.bytes)) {
+        return new Sk.builtin.bytes(...arguments);
+    }
+    Sk.builtin.pyCheckArgsLen("bytes", arguments.length, 0, 3);
+
+    if (errors === undefined) {
+        errors = "strict";
+    } else if (!(errors instanceof Sk.builtin.str)) {
+        throw new Sk.builtin.TypeError("bytes() argument 2 must be str, not " + Sk.abstr.typeName(errors));
+    } else {
+        errors = errors.v;
+    }
+    if (encoding === undefined) {
+        encoding = Sk.builtin.str.$utf8;
+    }
+    if (!(errors === "strict" || errors === "ignore" || errors === "replace")) {
+        throw new Sk.builtin.NotImplementedError("'" + errors + "' error handling not implemented in Skulpt");
+    }
+    if (arguments.length === 0) {
+        return new Sk.builtin.bytes(0);
+    }
+    if (arguments.length === 1) {
+        if (Sk.builtin.checkInt(source)) {
+            source = Sk.builtin.asnum$(source);
+            arr = new Uint8Array(source);
+        } else if (source instanceof Sk.builtin.bytes) {
+            return source;
+        } else if (Array.isArray(source)) {
+            // Internal fast path
+            Sk.asserts.assert(source.every((x) => (x >= 0) && (x < 256)),
+                              "Bad internal call to bytes with array object");
```

Comment:
what did you think about this btw? A worthwhile sanity check or a check that has zero probability of failing?

Reply:
Asserts are no-ops in production, so I don't think it hurts anything to have it here and can only help.